runtime/autoload/spellfile.lua

```diff
@@ -0,0 +1,137 @@
+local M = {}
```

Comment:
No need to expose all these functions (and in fact the configuration).

Also, you may want to wait for #29327.

Reply:
Ok, this sounds stupid, but beginners make stupid decisions and questions, so here we go : )

I exposed a lot of functions so I could write tests for them. In other words, I don't know how to write tests if I don't expose and `require` in the test file. Any ideas or references so I can get to a better design here?